modules/RTC/TPCUtils.js

```diff
@@ -339,21 +339,41 @@ export class TPCUtils {
 
     /**
      * Replaces the existing track on a RTCRtpSender with the given track.
+     *
      * @param {JitsiLocalTrack} oldTrack - existing track on the sender that needs to be removed.
      * @param {JitsiLocalTrack} newTrack - new track that needs to be added to the sender.
-     * @returns {Promise<void>} - resolved when done.
+     * @returns {Promise<RTCRtpTransceiver>} - resolved with the associated transceiver when done, rejected otherwise.
      */
     replaceTrack(oldTrack, newTrack) {
         const mediaType = newTrack?.getType() ?? oldTrack?.getType();
-        const transceiver = this.findTransceiver(mediaType, oldTrack);
         const track = newTrack?.getTrack() ?? null;
+        let transceiver;
+
+        // If old track exists, replace the track on the corresponding sender.
+        if (oldTrack) {
+            transceiver = this.pc.peerconnection.getTransceivers()?.find(t => t.sender?.track === oldTrack.getTrack());
+
+        // Find the first recvonly transceiver when a second track of the same media type is being added to the pc. As
+        // part of the track addition, a new m-line was added to the remote description with direction set to recvonly.
+        } else if (this.pc.getLocalTracks(mediaType)?.length && !newTrack?.conference) {
+            transceiver = this.pc.peerconnection.getTransceivers()?.find(
+                t => t.receiver?.track?.kind === mediaType
+                && t.direction === MediaDirection.RECVONLY
+                && t.currentDirection === MediaDirection.INACTIVE);
+
+        // If a track of a given media type is added for the first time, replace the track on the first sender. This
+        // will be called when the user joins the call muted but unmutes during the call.
```

Comment:
how do you know it's okay to just pick the first one in the multi track scenario? were we not supposed to generate dedicated mlines for each local track and re-use them as tracks are added/remove/replaced? I thought that the recvonly mlines will not be used for sending local tracks, but dedicate 'sendonly' mlines will be created?

Reply:
When we add a transceiver by adding a new m-line in the remote description without a ssrc attached, the browser will create a transceiver with the direction as RECVONLY since the local track hasn't been added yet and the current direction will be INACTIVE since there is no media flowing yet. Current direction is the differentiator here. After the track is added and reneg is done, the transceiver direction will change to sendonly.